File: notes_server/models/note.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Note struct {
	ID         string             `json:"id" bson:"_id,omitempty"`
	Slug       string             `json:"slug" bson:"slug"`
	CipherNote string             `json:"ciphernote" bson:"ciphernote"`
	UserID     primitive.ObjectID `json:"userId" bson:"userId"`
	CreatedAt  time.Time          `json:"createdAt" bson:"createdAt"`
	ExpiresAt  *time.Time         `json:"expiresAt,omitempty" bson:"expiresAt,omitempty"`
}

File: silentecho_app/src/model/Note.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface Note extends Document {
    userId: mongoose.Types.ObjectId;
    ciphernote: string;
    slug: string;
    expiresAt?: Date;
    createdAt: Date;
}

const NoteSchema: Schema<Note> = new mongoose.Schema({
    userId: {
        type: mongoose.Schema.Types.ObjectId,
        required: true,
        ref: 'User',
    },
    ciphernote: {
        type: String,
        required: true,
    },
    slug: {
        type: String,
        required: true,
        unique: true,
    },
    expiresAt: {
        type: Date,
        default: null,
    },
    createdAt: {
        type: Date,
        required: true,
        default: Date.now,
    },
});

const NoteModel =
    (mongoose.models.Note as mongoose.Model<Note>) ||
    mongoose.model<Note>('Note', NoteSchema);

export default NoteModel;
